src/pyflakes/plugins/string_dot_format.rs

```diff
@@ -0,0 +1,12 @@
+use rustpython_ast::Expr;
+
+use crate::ast::types::Range;
+use crate::check_ast::Checker;
+use crate::pyflakes::checks;
+
+/// F521
+pub fn string_dot_format_invalid(checker: &mut Checker, expr: &Expr, literal: &str) {
+    if let Some(check) = checks::string_dot_format_invalid(literal, Range::from_located(expr)) {
+        checker.add_check(check);
+    }
+}
```

Comment:
You could also choose to put this in `pyflakes/checks.rs` directly if it just returns an `Option<Check>`. (The undocumented convention is that `plugins` take `&mut Checker` while `checks` are pure functions that return checks.)

Reply:
Nice, will move them to the right place

Reply:
Assume you meant I can put this logic in `check_ast.rs` directly - moved it there. Let me know if that's not what you meant.